src/Func.h

```diff
@@ -414,6 +415,11 @@ class Func {
     /** The random seed to use for realizations of this function. */
     uint32_t random_seed;
 
+    /** Whether or not function schedules should be reset before
+     * performing any automatic scheduling. */
+    bool auto_schedule_reset;
```

Comment:
What's the motivation for ever resetting the user schedule? I could see some things getting accidentally silently dropped (e.g. .bound() directives)